packages/pigeon/test/pigeon_lib_test.dart

```diff
@@ -416,4 +416,26 @@ void main() {
     expect(parseResult.errors[0].message, contains('Generic fields'));
     expect(parseResult.errors[0].lineNumber, isNotNull);
   });
+
+  test('error with static field', () {
+    final Pigeon dartle = Pigeon.setup();
+    const String code = '''
+class WithStaticField {
+  static int? x;
+  int? y;
+}
+
+@HostApi()
+abstract class WithStaticFieldApi {
+  void doit(WithStaticField withTemplate);
+}
+''';
+    _withTempFile('compilationError.dart', (File file) {
+      file.writeAsStringSync(code);
```

Comment:
Is there a reason the content isn't an argument to the helper function? Every use starts with this line, and it's hard for me to imagine a scenario where you'd be using this helper and *not* immediately write contents to it.

Reply:
The `_withTempFile` function just generic.  It's something I've written a handful of times over the past years.  I can look into making a more apropos function for the tests.  I have like 3 dependent outstanding PRs that use it so it'd be better to look at it after they've landed.